astropy/coordinates/sky_coordinate.py

```diff
@@ -227,7 +230,7 @@ def _parse_inputs(self, args, kwargs):
                 # coord_kwargs will contain keys like 'ra', 'dec', 'distance'
                 # along with any frame attributes like equinox or obstime which
                 # were explicitly specified in the coordinate object (i.e. non-default).
-                coord_kwargs = _parse_coordinate_arg(args[0], frame, units)
+                coord_kwargs = _parse_coordinate_arg(args[0], frame, units, kwargs)
```

Comment:
Shouldn't this be `valid_kwargs`?  At this point `kwargs` has been verified to be empty as the result of popping off the valid kwargs.